code/modules/reagents/newchem/chem_heater.dm

```diff
@@ -0,0 +1,118 @@
+/obj/machinery/chem_heater
```

Comment:
future PR, every single codebase has special snowflake construction and effects, and I'll do a deeper analysis into how yours works in the future for this.